drift_dev/lib/src/writer/manager/table_manager_writer.dart

```diff
@@ -0,0 +1,212 @@
+// ignore_for_file: unnecessary_this
+
+part of "database_manager_writer.dart";
+
+class _TableManagerWriter {
+  /// The table that this manager is for
+  final DriftTable table;
+
+  /// Current generation scope
+  final Scope scope;
+
+  /// The name of the database class
+  ///
+  /// This does not contain any prefixes, use the [_templates] to get the correct name
+  /// to use in the generated code
+  final String dbClassName;
+
+  final List<DriftTable> otherTables;
+
+  _TableManagerWriter(
+      {required this.table,
+      required this.scope,
+      required this.dbClassName,
+      required this.otherTables})
+      : assert(table.existingRowClass == null,
+            "Manager Writer should ignore tables with custom row classes");
```

Comment:
I think it's fine for now, but we should investigate supporting custom row classes as well (#2989). I'm happy to take a look at that, do you recall what didn't work with custom classes?